src/routes/userRoutes.ts

```diff
@@ -4,12 +4,20 @@ import {
     createUserController,
     userLogin  } 
 from "../controllers/userControllers";
+import { 
+    emailValidation,
+    validateUserSignUp
+ } from "../middleware/emailValidator";
 
 const userRoutes = Router();
 
 userRoutes.get("/", fetchAllUsers);
 userRoutes.post('/login',userLogin);
-userRoutes.post("/register", createUserController)
+userRoutes.post("/register", 
+ emailValidation, 
+ validateUserSignUp, 
```

Comment:
not like this @yvanddniyo this function will accept schema as parameter like this `validateSchema(signUpSchema)`